config/import-control.xml

```diff
@@ -7,7 +7,18 @@
     <allow pkg="java.io"/>
     <allow pkg="java.util"/>
     <allow pkg="com.github.checkstyle.regression.data"/>
+    <allow pkg="com.github.checkstyle.regression.util"/>
+    <allow pkg="org.immutables"/>
```

Comment:
> Supposed there are 5 fields then the equals method would have 32 branches(2^5). That's horrible. 

I assume we got around this in main project using `equalsverifier` dependency.

> So I think it is not bad if an annotation processing tool could generate these for us

Let's keep `org.immutables` than, but how about we require no custom methods in the interface and require them to be in `utility` classes so we can keep code coverage on them? I dislike the idea of having code we can't be sure we are covering.